src/socket/tcp.rs

```diff
@@ -1170,6 +1147,46 @@ impl<'a> TcpSocket<'a> {
         }
 
         if let Some(ack_number) = repr.ack_number {
+            // TODO: When flow control is implemented,
+            // refractor the following block within that implementation
+
+            // Detect and react to duplicate ACKs by:
+            // 1. Check if duplicate ACK and change self.local_rx_dup_acks accordingly
+            // 2. If exactly 3 duplicate ACKs recived, set for fast retransmit
+            // 3. Update the last received ACK (self.local_rx_last_ack)
+            match self.local_rx_last_ack {
+                // Duplicate ACK if payload empty and ACK doesn't move send window ->
+                // Increment duplicate ACK count and set for retransmit if we just recived
+                // the third duplicate ACK
+                Some(ref mut last_rx_ack) if
+                    repr.payload.len() == 0 &&
+                    *last_rx_ack == ack_number => {
+                    // Increment duplicate ACK count
+                    let overflowed_indication_symbol = match self.local_rx_dup_acks.checked_add(1) {
+                        Some(n) => { self.local_rx_dup_acks = n; "" },
+                        None => "+"
+                    };
```

Comment:
I hate to nitpick, but this match statement could be replaced with a `if self.local_rx_dup_acks == u8::max_value() { "+" } else { "" }` in the debug statement below, and then we wouldn't have logic and presentation of debug logging mixed.

Reply:
Don't worry about nitpicking, I think that is a great way of maintaining a good code standard! And I like getting your feedback because it makes me think twice about how I would write something. Keep on nitpicking 😊